buildconfig/stubs/pygame/_sdl2/controller.pyi

```diff
@@ -1,25 +1,30 @@
 from typing import Dict, Mapping, Optional
 
-from pygame.joystick import Joystick
+from pygame.joystick import JoystickType
 
 def init() -> None: ...
 def get_init() -> bool: ...
 def quit() -> None: ...
 def set_eventstate(state: bool) -> None: ...
 def get_eventstate() -> bool: ...
+def update() -> None: ...
 def get_count() -> int: ...
 def is_controller(index: int) -> bool: ...
 def name_forindex(index: int) -> Optional[str]: ...
 
 class Controller:
     def __init__(self, index: int) -> None: ...
+    @property
+    def name(self) -> str: ...
+    @property
+    def id(self) -> int: ...
     def init(self) -> None: ...
     def get_init(self) -> bool: ...
     def quit(self) -> None: ...
     @staticmethod
-    def from_joystick(joy: Joystick) -> Controller: ...
+    def from_joystick(joy: JoystickType) -> Controller: ...
     def attached(self) -> bool: ...
-    def as_joystick(self) -> Joystick: ...
+    def as_joystick(self) -> JoystickType: ...
```

Comment:
What is the difference between Joystick and JoystickType?

Reply:
found the answer in joystick.pyi:
`# according to the current implementation, Joystick is a function that returns
# a JoystickType instance. In the future, when the C implementation is fixed to
# add __init__/__new__ to Joystick and it's exported directly, the typestubs
# here must be updated too`